### PR TITLE
bugfix: null-check

### PR DESCRIPTION
**Pull request by alexadrake**
added null checks around requests & responses; added test that fails without null-checking.

### CODE REVIEW

**Review by simplesagar — APPROVED** [review]

File: src/main/java/dev/speakeasyapi/sdk/utils/SpeakeasyInterceptor.java [hunk 1]
```diff
@@ -25,14 +31,34 @@ public SpeakeasyInterceptor(String apiKey, String serverUrl, boolean secureGrpc,
     @Override
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
         startTime = new Date();
+        try {
+            inputStream = new TeeInputStream(request.getInputStream(), requestOutputStream);
```
Comment (simplesagar) [inline src/main/java/dev/speakeasyapi/sdk/utils/SpeakeasyInterceptor.java:35]: does this transparently proxy from `InputStream` to `OutputStream` ?

**Review by simplesagar — APPROVED** [review]

**Comment by TristanSpeakEasy** [comment]
Funnily enough I have done a lot of similar changes in my own branch as  I work to bring the Java sdk up to parity with the go one for unmatched requests. Will merge this and use this as a base shouldn't be too dissimilar

File: src/main/java/dev/speakeasyapi/sdk/utils/SpeakeasyInterceptor.java (hunk 1 above)
Comment (alexadrake) [reply src/main/java/dev/speakeasyapi/sdk/utils/SpeakeasyInterceptor.java:35]: yes

**Comment by TristanSpeakEasy** [comment]
> Funnily enough I have done a lot of similar changes in my own branch as I work to bring the Java sdk up to parity with the go one for unmatched requests. Will merge this and use this as a base shouldn't be too dissimilar

I have for now not done the similar changes in my PR and left them for another pass, I was doing these to support setting an upper limit on the size of the request/response bodies that can be capture to match parity with the Go SDK, but I will pick that up once this is merged as its not a priority right now

**Review by TristanSpeakEasy — APPROVED** [review]

**Review by anuraagnalluri — APPROVED** [review]

**Pull request merged**